packages/stream/test/test-prime.js

```diff
@@ -0,0 +1,98 @@
+// @ts-check
+/* eslint-disable require-yield, no-empty-function */
+
+import 'ses';
+import './lockdown.js';
+
+import rawTest from 'ava';
+import { wrapTest } from '@endo/ses-ava';
+import { prime } from '../index.js';
+
+const test = wrapTest(rawTest);
+
+test('prime single next', async (/** @type {import('ava').ExecutionContext} */ t) => {
+  async function* single() {
+    t.is(yield, 0);
+  }
+
+  const iterator = prime(single());
+  const { done, value } = await iterator.next(0);
+  t.is(done, true);
+  t.is(value, undefined);
+});
+
+test('prime single return', async (/** @type {import('ava').ExecutionContext} */ t) => {
+  async function* empty() {
+    return 'Z';
+  }
+
+  const iterator = prime(empty());
+  const { done, value } = await iterator.return(0);
+  t.is(done, true);
+  t.is(value, 'Z');
+});
+
+test('prime empty throw in', async (/** @type {import('ava').ExecutionContext} */ t) => {
+  async function* empty() {
+    return 'Z';
+  }
+
+  const iterator = prime(empty());
+  const { done, value } = await iterator.throw(new Error('Abort'));
+  t.is(done, true);
+  t.is(value, 'Z');
```

Comment:
This seems somewhat surprising. I'm actually wondering if the default behavior shouldn't be to rethrow the error if the result of priming is a close (`done: true`).

Reply:
I concur. I’m following up with a change that causes throw to rethrow if the primer ran to the return.